digitalfabrik/integreat-app
# 2799: Implement chat api

### Short description

As a user i want to get information directly from a service user. Now the user can reach a service user that uses zammad help desk

### Proposed changes



- add get and post endpoint for chat
- add polling for message refresh
- add uuid creation for device id

### Side effects



- none
### Resolved issues



Fixes: #2799

### Testing
1. Start a chat (Testumgebung or Landeshauptstadt München)
2. Login to zammad helpdesk :https://zammad-integreat.tuerantuer.org/
3. Use credentials: https://passbolt.tuerantuer.org/app/passwords/view/a9e7cb57-2a92-453b-bb3d-c89fe8bbe1f9
4. Check for the deviceId in `localStorage` and look for the chat in zammad
5. Answer the chat message (message polling is 16s now)

Mind that showing a loading spinner can be problematic, since its reloading the chat with polling. Just showing loading spinner on initial loading
